pkg/prometheus/common_test.go

```diff
@@ -159,3 +160,125 @@ func TestBuildCommonPrometheusArgsWithRemoteWriteMessageV2(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildCommonPrometheusArgsWithOTLPReceiver(t *testing.T) {
+	for _, tc := range []struct {
+		version                    string
+		enableOTLPReceiver         *bool
+		expectedOTLPReceiverFlag   bool
+		OTLPConfig                 *monitoringv1.OTLPConfig
+		expectedOTLPFeatureEnabled bool
+	}{
+		// OTLP receiver not supported.
+		{
+			version:                    "2.46.0",
+			enableOTLPReceiver:         ptr.To(true),
+			expectedOTLPFeatureEnabled: false,
+			expectedOTLPReceiverFlag:   false,
+		},
+		// OTLP receiver supported starting with v2.47.0.
+		{
+			version:                    "2.47.0",
+			enableOTLPReceiver:         ptr.To(true),
+			expectedOTLPFeatureEnabled: true,
+			expectedOTLPReceiverFlag:   false,
+		},
+		// OTLP receiver supported but not enabled.
+		{
+			version:                    "2.47.0",
+			enableOTLPReceiver:         ptr.To(false),
+			expectedOTLPFeatureEnabled: false,
+			expectedOTLPReceiverFlag:   false,
+		},
+		// OTLP receiver config supported but verison not support
```

Comment:
```suggestion
		// OTLP receiver config supported but version not support
```

Reply:
Since this change comes from upstream, need to fix typo there